### PR TITLE
[CMake] Make PROOF optional and enabled by default

### PR DESCRIPTION
**Pull request by dpiparo**
# This Pull request:

## Changes or fixes:
Makes PROOF optional and enabled by default with some simple CMake changes.

## Checklist:

- [v] tested changes locally
- [ ] updated the docs (if necessary)

This PR fixes # 
There is no issue (yet) to be fixed, this is a PR opened for discussion. It is important to underline that effectively, this PR changes nothing: PROOF is still always built by default and the opt-out option is given not to build it.

### CODE REVIEW

**Comment by Axel-Naumann** [comment]
Thanks!

FYI apart from some weird Jenkins errors we get:
```
2023-08-30T18:30:54.7170667Z [2871/3458] Building CXX object test/CMakeFiles/stressProof.dir/stressProof.cxx.o
2023-08-30T18:30:54.7290088Z FAILED: test/CMakeFiles/stressProof.dir/stressProof.cxx.o 
2023-08-30T18:30:54.7394177Z /usr/bin/c++  -I/github/home/ROOT-CI/build/include -I/github/home/ROOT-CI/src/proof/proof/inc -I/github/home/ROOT-CI/src/math/mathcore/v7/inc -I/github/home/ROOT-CI/src/math/mathcore/inc -I/github/home/ROOT-CI/build/ginclude -I/github/home/ROOT-CI/src/core/base/inc -I/github/home/ROOT-CI/src/core/base/v7/inc -I/github/home/ROOT-CI/src/core/clib/inc -I/github/home/ROOT-CI/src/core/clingutils/inc -I/github/home/ROOT-CI/src/core/clingutils/res -I/github/home/ROOT-CI/src/core/cont/inc -I/github/home/ROOT-CI/src/core/foundation/inc -I/github/home/ROOT-CI/src/core/foundation/v7/inc -I/github/home/ROOT-CI/src/core/gui/inc -I/github/home/ROOT-CI/src/core/meta/inc -I/github/home/ROOT-CI/src/core/textinput/inc -I/github/home/ROOT-CI/src/core/thread/inc -I/github/home/ROOT-CI/src/core/zip/inc -I/github/home/ROOT-CI/src/core/lzma/inc -I/github/home/ROOT-CI/src/core/lz4/inc -I/github/home/ROOT-CI/src/core/zstd/inc -I/github/home/ROOT-CI/src/builtins/zstd -I/github/home/ROOT-CI/src/core/imt/inc -I/github/home/ROOT-CI/src/core/multiproc/inc -I/github/home/ROOT-CI/src/net/net/inc -I/github/home/ROOT-CI/src/io/io/v7/inc -I/github/home/ROOT-CI/src/io/io/inc -I/github/home/ROOT-CI/src/tree/tree/inc -I/github/home/ROOT-CI/src/proof/proofplayer/inc -I/github/home/ROOT-CI/src/hist/hist/inc -I/github/home/ROOT-CI/src/math/matrix/inc -I/github/home/ROOT-CI/src/tree/treeplayer/inc -I/github/home/ROOT-CI/src/graf2d/gpad/inc -I/github/home/ROOT-CI/src/graf2d/graf/inc -I/github/home/ROOT-CI/src/graf3d/g3d/inc -fdiagnostics-color=always -std=c++20 -Wno-implicit-fallthrough -Wno-noexcept-type -pipe  -Wshadow -Wall -W -Woverloaded-virtual -fsigned-char -pthread -DDNNCPU -g -std=c++20 -MD -MT test/CMakeFiles/stressProof.dir/stressProof.cxx.o -MF test/CMakeFiles/stressProof.dir/stressProof.cxx.o.d -o test/CMakeFiles/stressProof.dir/stressProof.cxx.o -c /github/home/ROOT-CI/src/test/stressProof.cxx
2023-08-30T18:30:54.7559009Z ##[error]/github/home/ROOT-CI/src/test/stressProof.cxx:140:10: fatal error: proof/getProof.C: No such file or directory
2023-08-30T18:30:54.7654053Z   140 | #include "proof/getProof.C"
2023-08-30T18:30:54.7654343Z       |          ^~~~~~~~~~~~~~~~~~
2023-08-30T18:30:54.7654778Z compilation terminated.
```

**Comment by Axel-Naumann** [comment]
I have updated the root-spi repo for the old (Jenkins) CI to learn about this new module and know that it should be built everywhere.

**Comment by Axel-Naumann** [comment]
@phsft-bot build

**Comment by dpiparo** [comment]
@phsft-bot build just on mac12arm/cxx20, mac11/noimt, ROOT-ubuntu2004/python3, ROOT-ubuntu2204/nortcxxmod

**Comment by dpiparo** [comment]
@phsft-bot build just on mac12arm/cxx20, mac11/noimt, ROOT-ubuntu2004/python3, ROOT-ubuntu2204/nortcxxmod

**Comment by dpiparo** [comment]
@phsft-bot build

**Comment by dpiparo** [comment]
Ok, apart from TMVA tests, it seems that the changes to CMake have the desired effect.

**Comment by dpiparo** [comment]
@phsft-bot build

**Comment by dpiparo** [comment]
Ok, this is tmva hitting the testing phase. The changes work as expected.

**Comment by hahnjo** [comment]
Just for my curiosity because it's not explained in the commit message: You are adding `include_directories(${ROOT_SOURCE_DIR}/tutorials)` in `test/CMakeLists.txt`, I suppose this is to fix https://github.com/root-project/root/pull/13571#issuecomment-1700410866. I was expecting that this happens when PROOF is disabled - where is that directory added before / otherwise?

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -29,6 +29,7 @@ endif()
 if(DEFINED ROOT_SOURCE_DIR)
   # Testing using the binary tree
   set(ROOT_root_CMD ${ROOT_BINDIR}/root.exe)
+  include_directories(${ROOT_SOURCE_DIR}/tutorials)
```
Comment (pcanal) [inline test/CMakeLists.txt:32]: Why is this addition unconditional?  Why do we need to add something when disabling something else (`PROOF`)?

**Pull request merged**